tests/Doctrine/Tests/OrmFunctionalTestCase.php

```diff
@@ -946,4 +950,60 @@ final protected function getLastLoggedQuery(int $index = 0): array
 
         return $lastQuery;
     }
+
+    public function getLimitSQLByPlatform($limit, AbstractPlatform $platform): string
+    {
+        $sql = ' LIMIT';
+        if ($platform instanceof OraclePlatform) {
+            $sql = ' WHERE ROWNUM <=';
+        }
+
+        return $sql . ' ' . $limit;
+    }
+
+    public function getDiscriminatorByPlatform($key, AbstractPlatform $platform): string
+    {
+        if ($platform instanceof OraclePlatform || $platform instanceof DB2Platform) {
+            return strtoupper($key);
+        }
+
+        return $key;
+    }
+
+    /**
+     * Due to autocommit in oracle database, all data in temporary tables for the current session is dropped
```

Comment:
What are the temporary tables? Why isn't this necessary in the DBAL test suite?

Reply:
`CREATE GLOBAL TEMPORARY TABLE`

\Doctrine\DBAL\Platforms\OraclePlatform::getCreateTemporaryTableSnippetSQL